mlflow/server/__init__.py

```diff
@@ -246,6 +247,50 @@ def _build_gunicorn_command(gunicorn_opts, host, port, workers, app_name):
     ]
 
 
+def _build_uvicorn_command(uvicorn_opts, host, port, workers, app_name):
+    """
+    Build command to run uvicorn server.
+
+    Uses gunicorn with uvicorn workers for better process management:
+    - Automatic worker restart on failure
+    - Better resource management
+    - More robust production deployment
+    """
+    bind_address = f"{host}:{port}"
+    opts = shlex.split(uvicorn_opts) if uvicorn_opts else []
+
+    # On Windows, use pure uvicorn since gunicorn is not supported
+    if is_windows():
+        return [
+            sys.executable,
+            "-m",
+            "uvicorn",
+            *opts,
+            "--host",
+            host,
+            "--port",
+            str(port),
+            "--workers",
+            str(workers),
+            app_name,
+        ]
+
+    # On Unix systems, use gunicorn with uvicorn workers for better process management
+    return [
+        sys.executable,
+        "-m",
+        "gunicorn",
+        "-k",
+        "uvicorn_worker.UvicornWorker",
```

Comment:
Does `uvicorn` require `uvicorn_worker`? Should we add it as an mlflow dependency?